app/controllers/admin/creator_settings_controller.rb

```diff
@@ -1,15 +1,26 @@
 module Admin
   class CreatorSettingsController < Admin::ApplicationController
-    ALLOWED_PARAMS = %i[community_name logo_svg primary_brand_color_hex invite_only_mode public checked_code_of_conduct
-                        checked_terms_and_conditions].freeze
+    ALLOWED_PARAMS = %i[community_name primary_brand_color_hex invite_only_mode public checked_code_of_conduct
+                        checked_terms_and_conditions logo].freeze
 
-    def new; end
+    def new
+      @max_file_size = LogoUploader::MAX_FILE_SIZE
+      @logo_allowed_types = (LogoUploader::CONTENT_TYPE_ALLOWLIST +
+        LogoUploader::EXTENSION_ALLOWLIST.map { |extension| ".#{extension}" }).join(",")
+    end
 
     def create
       extra_authorization
+
+      raise CarrierWave::IntegrityError if settings_params[:logo].blank?
+
       ActiveRecord::Base.transaction do
+        logo_uploader = upload_logo(settings_params[:logo])
```

Comment:
Given that we're wrapping this in a transaction, consider moving the logo uploader later in the block.  There are a few reasons around transactionality.  Namely, I suspect that the logo upload is the expensive work to do.  But, if something later fails (albeit unlikely) we will have done the expensive work and there are aspects that we can't rollback (e.g. the compute and creation of images).

Reply:
Sounds sensible, I'll update this.

Reply:
✅  done